### DISCUSSION TITLE
Browser support 5.0

### DISCUSSION THREAD

**Discussion by waterplea** [taiga-family/taiga-ui#3232] [RFC]
Here I will list all desired Web APIs and their browser support to have a rough idea of what browser support do we want for the next major release.

| Browser name    | Current | Forecast | Description                                                                                                                        |
|-----------------|---------|----------|------------------------------------------------------------------------------------------------------------------------------------|
| Chrome Mobile   | 88+     |          |                                                                                                                                    |
| Chrome Desktop  | 88+     |          |                                                                                                                                    |
| Edge Desktop    | 88+     |          |                                                                                                                                    |
| Yandex Desktop  | 21.2+   |          |                                                                                                                                    |
| Yandex Mobile   | 21.2+   |          |                                                                                                                                    |
| Opera Desktop   | 74+     |          |                                                                                                                                    |
| Opera Mobile    | 63+     |          |                                                                                                                                    |
| Safari Mobile   | 13.4+   | 14.5+    | [flex gap](https://caniuse.com/flexbox-gap), [inset](https://developer.mozilla.org/en-US/docs/Web/CSS/inset#browser_compatibility), [BigInt](https://caniuse.com/bigint), [:is()](https://caniuse.com/css-matches-pseudo), [padding-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/padding-inline#browser_compatibility), [margin-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-inline#browser_compatibility), [border-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/border-inline#browser_compatibility), [inset-inline-start](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-inline-start#browser_compatibility), [inset-inline-end](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-inline-end#browser_compatibility), [inset-block-start](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-block-start#browser_compatibility),  [inset-block-end](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-block-end) |
| Safari Desktop  | 13.1+   | 14.1+    | [flex gap](https://caniuse.com/flexbox-gap), [inset](https://developer.mozilla.org/en-US/docs/Web/CSS/inset#browser_compatibility), [BigInt](https://caniuse.com/bigint), [:is()](https://caniuse.com/css-matches-pseudo), [padding-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/padding-inline#browser_compatibility), [margin-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-inline#browser_compatibility), [border-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/border-inline#browser_compatibility), [inset-inline-start](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-inline-start#browser_compatibility), [inset-inline-end](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-inline-end#browser_compatibility), [inset-block-start](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-block-start#browser_compatibility),  [inset-block-end](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-block-end) <br /> [\<input type="datetime-local">](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#browser_compatibility) |
| Samsung Mobile  | 15+     |          |                                                                                                                                    |
| Firefox Mobile  | 120+    |    121+      |                                                                                                                                    |
| Firefox Desktop | 120+    |   121+     |                                                                                                                                    |

___

- ✔ [flexbox gap](https://developer.mozilla.org/en-US/docs/Web/CSS/gap) 
**Safari Desktop:** 14.1+ | **Safari IOS:** 14.5+
- ✔ [inset](https://developer.mozilla.org/en-US/docs/Web/CSS/inset#browser_compatibility)
**Safari Desktop:** 14.1+ | **Safari IOS:** 14.5+
- ✔ [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) 
**Safari Desktop:** 14+ | **Safari IOS:** 14+
- ✔ [:is()](https://caniuse.com/css-matches-pseudo)
**Safari Desktop:** 14+ | **Safari IOS:** 14+
- ✔ [\<input type="datetime-local">](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#browser_compatibility)
**Safari Desktop:** 14.1+
- ✔ [padding-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/padding-inline#browser_compatibility) / [margin-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-inline#browser_compatibility) / [border-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/border-inline#browser_compatibility)
**Safari Desktop:** 14.1+ | **Safari IOS:** 14.5+

<br />

- ❌ [FocusVisible](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) 
**Chrome** 86, **Safari** 15.4, **Firefox** 85
- ❌ [:has()](https://caniuse.com/css-has)
**Chrome:** 105+ | **Safari Desktop:** 15.4+ | **Safari IOS:** 15.4+
- ❌ [<dialog />](https://caniuse.com/dialog)
**Safari Desktop:** 15.4+ | **Safari IOS:** 15.4+
- ❌ [aspect-ratio](https://developer.mozilla.org/en-US/docs/Web/CSS/aspect-ratio)
**Safari Desktop:** 15+ | **Safari IOS:** 15+
- ❌ [border-start-start-radius](https://developer.mozilla.org/en-US/docs/Web/CSS/border-start-start-radius)
**Safari Desktop:** 15+ | **Safari IOS:** 15+

**Comment by nsbarsukov** [comment]
- [Promise.finally](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally)
**Firefox** 58

**Reply by nsbarsukov** [reply]
https://github.com/taiga-family/taiga-ui/blob/78b8e921482793c35f04e1501397b4ea74f273bd/projects/addon-doc/src/components/example/example.component.ts#L120

**Comment by nsbarsukov** [comment]
- [Object.fromEntries](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries)
**Firefox** 63

**Reply by nsbarsukov** [reply]
https://github.com/taiga-family/taiga-ui/blob/5f65b72eddd23882318d598558524d32bbb3ecd8/projects/cdk/utils/miscellaneous/object-from-entries.ts#L1-L9

**Comment by nsbarsukov** [comment]
* [String.prototype.replaceAll](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll)

**Chrome** 85+
**Safari IOS** 13.4+

**Comment by nsbarsukov** [comment]
* [Optional chaining operator (?.)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)
* [Nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)

**Safari Desktop** 13.1+
**Safari IOS** 13.4+

**Comment by nsbarsukov** [comment]
Bump for **4.0**

* https://github.com/taiga-family/taiga-ui/pull/6818

**Comment by nsbarsukov** [comment]
* [gap](https://caniuse.com/flexbox-gap) (flex)
* [inset](https://developer.mozilla.org/en-US/docs/Web/CSS/inset#browser_compatibility)

**Safari Desktop:** 14.1+
**Safari IOS:** 14.5+

**Comment by nsbarsukov** [comment]
- [BigInt](https://caniuse.com/bigint)
- [:is()](https://caniuse.com/css-matches-pseudo)

**Safari Desktop:** 14+
**Safari IOS:** 14+

**Comment by nsbarsukov** [comment]
- [:has()](https://caniuse.com/css-has)

**Chrome:** 105+
**Safari Desktop:** 15.4+
**Safari IOS:** 15.4+

**Comment by nsbarsukov** [comment]
- [<dialog />](https://caniuse.com/dialog)

**Safari Desktop:** 15.4+
**Safari IOS:** 15.4+

**Comment by nsbarsukov** [comment]
- [\<input type="datetime-local">](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#browser_compatibility)

**Safari Desktop:** 14.1+

**Comment by nsbarsukov** [comment]
* [padding-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/padding-inline#browser_compatibility)
* [margin-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-inline#browser_compatibility)
* [border-inline](https://developer.mozilla.org/en-US/docs/Web/CSS/border-inline#browser_compatibility)
* [inset-inline-start](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-inline-start#browser_compatibility) / [inset-inline-end](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-inline-end#browser_compatibility) / [inset-block-start](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-block-start#browser_compatibility) /  [inset-block-end](https://developer.mozilla.org/en-US/docs/Web/CSS/inset-block-end)

**Safari Desktop**: 14.4+
**Safari IOS**: 14.5+

**TODO**: don't forget to [update stylelint](https://github.com/taiga-family/configurations/blob/0cf266623043e5160c14b29929502619777b32eb/projects/stylelint-config/index.js#L165-L168)!

**Comment by waterplea** [comment]
- [aspect-ratio](https://developer.mozilla.org/en-US/docs/Web/CSS/aspect-ratio)
- [border-start-start-radius](https://developer.mozilla.org/en-US/docs/Web/CSS/border-start-start-radius)

**Safari:** 15+

**Comment by nsbarsukov** [comment]
https://github.com/taiga-family/taiga-ui/releases/tag/v5.0.0
